compiler/aarch64/codegen/BinaryEvaluator.cpp

```diff
@@ -38,10 +76,27 @@ OMR::ARM64::TreeEvaluator::laddEvaluator(TR::Node *node, TR::CodeGenerator *cg)
 
 TR::Register *
 OMR::ARM64::TreeEvaluator::isubEvaluator(TR::Node *node, TR::CodeGenerator *cg)
-	{
-	// TODO:ARM64: Enable TR::TreeEvaluator::isubEvaluator in compiler/aarch64/codegen/TreeEvaluatorTable.hpp when Implemented.
-	return OMR::ARM64::TreeEvaluator::unImpOpEvaluator(node, cg);
-	}
+   {
+   TR::Node *firstChild = node->getFirstChild();
+   TR::Register *src1Reg = cg->evaluate(firstChild);
+   TR::Node *secondChild = node->getSecondChild();
+   TR::Register *trgReg = cg->allocateRegister();
+
+   if (secondChild->getOpCode().isLoadConst() && secondChild->getRegister() == NULL)
+      {
+      addConstant32ToInteger(node, trgReg, src1Reg, -secondChild->getInt(), cg);
```

Comment:
AArch64 has an immediate form subtract instruction that does this kind of operation internally.  What is the advantage of manually converting the instruction and constant operand into an addition?

Reply:
I thought it would save the code size of the JIT compiler itself.
I changed the function `addConstant32ToInteger()`  to `addOrSubConstant32ToInteger()` that takes an extra parameter.